### DISCUSSION TITLE
Safely setting service startup type to "Automatic (Delayed Start)"

### DISCUSSION THREAD

**Discussion by Tyler-Duckworth** [wixtoolset/issues#8721] [Questions]
Hi!

I'm in the process of creating a WiX v4 installer to install a Windows Service with a default Startup Type of "Automatic (Delayed Start)". I have been able to get it working using the `DelayedAutoStart` attribute in the (non-util) `ServiceConfig` element ([docs](https://wixtoolset.org/docs/schema/wxs/serviceconfig/)), but it does result in a WIX1149 warning about the functionality not working correctly in MSI 5.0 that recommends `util:ServiceConfig` ([docs](https://wixtoolset.org/docs/schema/util/serviceconfig/)) instead. 

However, I do not see any fields in `util:ServiceConfig` related to the starting of the service. Is there a way in WiX to do this without potentially using this incomplete functionality? (I've considered doing a custom action that calls `sc.exe` but didn't know if there was a more 'recommended' way to do it)

**Comment by Tyler-Duckworth** [comment]
Hey! I wanted to comment on this and see if anyone had been able to see this post at all. It's been a little over a month since I posted this and so far I haven't been able to find out a recommended way to set up the delayed start to a Windows Service in WiX.

**Reply by Tyler-Duckworth** [reply]
@robmen, do you know if there's a way you would recommend handling this warning?

Apologies for the @, but I posted this a while ago and haven't gotten any response from anyone.

**Reply by robmen** [reply]
Yeah, I'm busy with customers at this time. Other volunteers get busy for lots of different reasons. If you want guaranteed answers, [be a customer](https://www.firegiant.com/services/).

**Reply by Tyler-Duckworth** [reply]
> Yeah, I'm busy with customers at this time. Other volunteers get busy for lots of different reasons. If you want guaranteed answers, [be a customer](https://www.firegiant.com/services/).

Hi. I wanted to touch base with this, since it's been around 6 months since I posted the initial question. 

Does anyone know of a recommended way to create a Service with an "Automatic (Delayed Start)" without causing any incomplete functionality?

**Comment by yerald231ger** [comment]
Hi Tyler. How did you get install a project as a service? I've been searching documentations about it, but i can't find use full info.

**Reply by Tyler-Duckworth** [reply]
There are a few ways. Here are some good tutorials I found:
- Rob Mensching (one of the main devs), has a YouTube video for how to do it with v4 ([source](https://www.youtube.com/watch?v=vZRZeDOTPZQ))
- There is a tutorial on the old v3 docs ([source](https://wixtoolset.org/docs/v3/howtos/general/install_windows_service/)). 

There doesn't seem to be a dedicated page on Windows Services with WiX on the new docs other than API documentation for the tags you can use. Hope this helps!

**Comment by felipecrs** [comment]
@Tyler-Duckworth I have no relationship with this project but found your question when Google-searching.

I believe [ServiceConfig](https://docs.firegiant.com/wix3/xsd/wix/serviceconfig/) is the ideal way to handle it.

```xml
<!--
Source - https://stackoverflow.com/a/9158372
Posted by Matt Sharpe, modified by community. See post 'Timeline' for change history
Retrieved 2025-11-12, License - CC BY-SA 4.0
-->

<ServiceConfig DelayedAutoStart="yes" OnInstall="yes" OnReinstall ="yes" />

```

_Taken from https://stackoverflow.com/a/9158372/12156188._
